src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -650,10 +650,11 @@ db::DataBox<tmpl::list<Tags...>>::reset_compute_items_after_mutate(
   using current_tags_to_reset = tmpl::list<TagsOfImmutableItemsToReset...>;
   using next_compute_tags_to_reset = tmpl::list_difference<
       tmpl::remove_duplicates<tmpl::transform<
-          tmpl::append<
-              tmpl::filter<typename DataBox<tmpl::list<Tags...>>::edge_list,
-                           std::is_same<tmpl::pin<TagsOfImmutableItemsToReset>,
-                                        tmpl::get_source<tmpl::_1>>>...>,
+          tmpl::filter<
+              typename DataBox<tmpl::list<Tags...>>::edge_list,
+              tmpl::bind<tmpl::list_contains,
```

Comment:
[optional] I recently added a `tmpl::lazy::list_contains`, if you want to avoid the `tmpl::bind`.  I don't know whether bind has a performance impact.

Reply:
I compiled with this change it seemed to be the same or ever so slightly less RAM (but the difference could just be noise), so I just threw it in